### DISCUSSION TITLE
Need to filter Certain Sites with which contains "-test"  from the search results.

### DISCUSSION THREAD

**Discussion by vikramsamal** [microsoft-search/pnp-modern-search#3179] [Q&A] [answered]
I have a requirement where we don't want to show results from paths/sites where the URL containing "-test" will be there. Can we add some filter criteria to block displaying such results from those sites? Any leads will be appreciated.

**Comment by MartinM85** [comment, marked as answer]
To exclude something, add `-` before `path`.

`-path:\"https://contoso.sharepoint.com/sites/*-test\"`

**Reply by wobba** [reply]
Except prefix wildcard is not supported in KQL. And the hyphen may likely be tokenized away. I know `path` is a bit special so interesting to see if it works or not.

**Reply by MartinM85** [reply]
I've tested it in the [Graph API](https://learn.microsoft.com/en-us/graph/api/resources/search-api-overview?view=graph-rest-1.0) which has support for KQL and it worked. But I can't say how Graph API search works under the hood.

**Comment by vikramsamal** [comment]
@MartinM85 Thanks for the help! It works. I could also find another way wanted to share if someone wants to use other way

`NOT spsiteurl:'*-test'`
